### PR TITLE
test: refactor old tests and cleanup test helpers

### PR DESCRIPTION
**Pull request by BioPhoton**
**Precondition:**
- #871 
- #872 

This PR includes: 
- a lot of changed files 😇
- align all tmp paths
- refactored the tests of `cli-e2e`
- uses nx-verdaccio
- replaced plugins with dummy
- deleted old testing code

### CODE REVIEW

File: e2e/cli-e2e/project.json [hunk 1]
```diff
@@ -11,22 +11,23 @@
...
-    "models",
-    "utils",
-    "core",
-    "cli",
-    "plugin-eslint",
-    "plugin-coverage",
-    "plugin-js-packages",
-    "react-todos-app"
-  ],
+  "implicitDependencies": ["cli", "cli-e2e-env"],
```
Comment (matejchalk) [inline e2e/cli-e2e/project.json:31]: Same feedback as for #871 and #872 - the environment root could just be sub-folder in `cli-e2e` for better colocation.

File: e2e/cli-e2e/tests/help.e2e.test.ts [hunk 2]
```diff
@@ -14,12 +18,13 @@ describe('CLI help', () => {
 
   it('should produce the same output to stdout for both help argument and help command', async () => {
     const helpArgResult = await executeProcess({
-      command: 'code-pushup',
-      args: ['help'],
+      command: 'npx',
+      args: ['@code-pushup/cli', 'help'],
     });
```
Comment (matejchalk) [inline e2e/cli-e2e/tests/help.e2e.test.ts:23]: You don't set `cwd` here, I'm guessing that's a mistake?

Comment (BioPhoton) [reply e2e/cli-e2e/tests/help.e2e.test.ts:23]: Yes!! Thx nice catch!

File: .github/workflows/code-pushup.yml [hunk 3]
```diff
@@ -36,4 +36,4 @@ jobs:
       - name: Run Code PushUp action
         uses: code-pushup/github-action@v0
         with:
-          bin: npx nx code-pushup --
+          bin: npx nx code-pushup -- --verbose
```
Comment (BioPhoton) [inline .github/workflows/code-pushup.yml:39]:
```suggestion
          bin: npx nx code-pushup --
```

File: e2e/plugin-coverage-e2e/tests/collect.e2e.test.ts [hunk 4]
```diff
@@ -2,18 +2,25 @@ import { cp } from 'node:fs/promises';
...
+  omitVariableReportData,
+} from '@code-pushup/test-utils';
 import { executeProcess, readJsonFile } from '@code-pushup/utils';
 
-describe('collect report with coverage-plugin NPM package', () => {
-  const envRoot = 'tmp/e2e/plugin-coverage-e2e';
-  const fixtureDir = join('e2e', 'plugin-coverage-e2e', 'mocks', 'fixtures');
-  const basicDir = join(envRoot, 'basic-setup');
-  const existingDir = join(envRoot, 'existing-report');
+describe('PLUGIN collect report with coverage-plugin NPM package', () => {
```
Comment (matejchalk) [inline e2e/plugin-coverage-e2e/tests/collect.e2e.test.ts:14]: Renaming the `describe` block means there is now a new snapshot, but the old one wasn't deleted. Not sure what's the benefit of prefixing it with `PLUGIN `. Either way, please remove obsolete snapshots with `npx nx nxv plugin-coverage-e2e -- --update`.

Same goes for `plugin-eslint-e2e` and `plugin-lighthouse-e2e`.

File: e2e/plugin-lighthouse-e2e/tests/collect.e2e.test.ts [hunk 5]
```diff
@@ -1,45 +1,50 @@
...
-  const baseDir = 'tmp/e2e/plugin-lighthouse-e2e/__test__';
+  const fixturesDir = join('e2e', nxTargetProject(), 'mocks/fixtures');
+  beforeAll(async () => {
+    await cp(fixturesDir, testFileDir, { recursive: true });
+  });
 
-  afterEach(async () => {
-    await teardownTestFolder(baseDir);
+  afterAll(async () => {
+    // await teardownTestFolder(testFileDir);
```
Comment (matejchalk) [inline e2e/plugin-lighthouse-e2e/tests/collect.e2e.test.ts:30]: Leftover from debugging?

```suggestion
    await teardownTestFolder(testFileDir);
```

File: static-environments/cli-e2e-env/package.json
Comment (matejchalk) [inline static-environments/cli-e2e-env/package.json:1]: I guess this file can now be deleted.

Comment (BioPhoton) [reply static-environments/cli-e2e-env/package.json:1]: Yes!

Comment (BioPhoton) [reply static-environments/cli-e2e-env/package.json:1]: deleted

**Review by matejchalk — DISMISSED** [review]

**Review by matejchalk — DISMISSED** [review]

**Pull request merged**